src/profiler.js

```diff
@@ -0,0 +1,61 @@
+// profiler.js
+
+var _ = require('underscore');
+
+/**
+ * This RDF Pipeline component profiler gathers metrics on the processing time for each
+ * component and records them in the specific metrics directory.  The sequence of execution
+ * is also logged if createLog is called to enable logging. 
+ *
+ * @param node the RDF pipeline component facade
+ */
+module.exports = function(node) { 
+
+    _.extend(node, 
+        {
+          profiler: {
+              metrics: {
+                  averageUpdateTime: 0,
+                  numberOfErrors: 0,
+                  numberOfEvents: 0,
+                  numberOfUpdates: 0,
+                  startTime: Date.now(),
+                  totalErrorTime: 0,
+                  totalProcessingTime: 0,
+                  totalUpdateTime: 0
+              },
+
+              eventTypes: {
```

Comment:
I like this idea, but I don't want non-stop profile calls in the framework which is already a little long.  Let met take a look at the code and see what we can do here.